### ISSUE TITLE
Keywords plugin overrides existing fields

### ISSUE THREAD

**Issue by zaim** [tblobaum/mongoose-troop#18]
If a source field already exists in the schema, the plugin will override it, meaning any custom validations, etc. on that field will be removed.

``` javascript
var schema = new Schema({
   title : {
    type : String,
    required : true
  }
});

schema.plugin(keywords, { source : 'title' });

// This will fail:
assert(schema.paths.title.options.required === true);
```

**Comment by tblobaum** [comment]
I tested this with mongoose 2.4.10 and 2.7.2 and in both cases the schema is not overridden, and `schema.paths.title.options.required` is true

You can see here the field is only added to the schema if it does not already exist https://github.com/tblobaum/mongoose-troop/blob/master/lib/keywords/index.js#L25-30

**Comment by zaim** [comment]
But isn't `source` an array? Wouldn't `schema.paths[source]` be incorrect? This somehow fails for me. I'm using mongoose 2.7.2.

**Comment by tblobaum** [comment]
Hmm.

``` js
> var troop = require('mongoose-troop')
undefined
> var schema = new mongoose.Schema({
...    title : {
.....     type : String,
.....     required : true
.....   }
... });
undefined
> schema.plugin(troop.keywords, { source : 'title' });
{ paths: 
   { title: 
      { enumValues: [],
        regExp: null,
        path: 'title',
        instance: 'String',
        validators: [Object],
        setters: [],
        getters: [],
        options: [Object],
        _index: null,
        isRequired: true },
     _id: 
      { path: '_id',
        instance: 'ObjectID',
        validators: [],
        setters: [Object],
        getters: [],
        options: [Object],
        _index: null,
        defaultValue: [Function] },
     keywords: 
      { casterConstructor: [Function: SchemaString],
        caster: [Object],
        path: 'keywords',
        instance: undefined,
        validators: [],
        setters: [],
        getters: [],
        options: [Object],
        _index: null,
        defaultValue: [Function] } },
  virtuals: { id: { getters: [Object], setters: [], options: {} } },
  nested: {},
  inherits: {},
  callQueue: [ [ 'pre', [Object] ] ],
  _indexes: [],
  methods: { extractKeywords: [Function] },
  statics: { extractKeywords: [Function] },
  tree: 
   { title: { required: true, type: [Function] },
     _id: { auto: true, type: [Function] },
     id: { getters: [Object], setters: [], options: {} },
     keywords: [ [Function] ] },
  options: { strict: false, safe: true } }
> schema.paths.title.options.required
true
> 
```

```

├─┬ mongoose@2.7.2 
│ ├── hooks@0.2.1 
│ └─┬ mongodb@1.0.2 
│   └── bson@0.0.6 
├─┬ mongoose-troop@0.0.6 
  ├── bcrypt@0.5.0 
  ├─┬ directory@0.0.61 
    └─┬ findit@0.1.2 
      └─┬ seq@0.3.5 
        ├─┬ chainsaw@0.0.9 
        │ └── traverse@0.3.9 
        └─┬ hashish@0.0.4 
          └── traverse@0.6.3 
```

**Comment by zaim** [comment]
It only fails if there is more than one item in the `source` array since using an array as a key for an object casts it to a string. E.g.

``` js
var a = ['one', 'two'], b = {};
b[a] = 'test';
console.log(b);    // prints { 'one,two': 'test' }
```

I fixed this in zaim@6eb746c4f8f31bbf4c6b542f8e26489b13055111

**Comment by tblobaum** [comment]
That makes sense, I'll pull in your fix.

Thanks
